repopulate trackModels when library dir is removed

kain88-de opened the pull request:
[FIX:1258589](https://bugs.launchpad.net/mixxx/+bug/1258589)

The problem was that the trackModels weren't update about the changes
that happend in the basetrackcache and the sqldatabase.

Is there a way that trackDAO could also inform the trackModels about changes (removal of data) itself like it does for the BaseTrackCache. Currently we make a change in TrackDAO and inform affected classes (BTC and the TrackModels) in several places from this change.

Commit:
repopulate trackModels when library dir is removed

FIX:1258589

The problem was that the trackModels weren't update about the changes
that happend in the basetrackcache and the sqldatabase.

ywwg commented on src/library/library.h:
```diff
@@ -95,6 +95,7 @@ class Library : public QObject {
   signals:
     void showTrackModel(QAbstractItemModel* model);
     void switchToView(const QString& view);
+    void updateTrackModel();
```
I'm not sure about this naming if this isn't emitted every time the track model is updated.  Is there a more specific name we can give this, or are there more places where we should be emitting?

kain88-de replied:
we should do this everytime tracks are removed from trackDAO. Afaik only then there is a chance that we have stale data in a trackModel.  Would `forceTrackModelUpdate` be a better name for you?

ywwg replied:
yeah that works for me

rryan commented:
It's a little weird that the signal makes it to the track models by way of the widget system. If we have multiple library widgets (which we don't really want to but do/did in LateNight) then the signal gets delivered N times. 

What if you connected the signal from TrackDAO to BaseSqlTableModel::select() in BaseSqlTableModel's constructor?

Commit:
Revert "repopulate trackModels when library dir is removed"

This reverts commit f8f53cfa97c405edbc042b83d3e2d3d307dec47e.

I will do this later with the BaseTrackModel and the TrackDAO directly.

Commit:
Notify BSTM from TrackDAO if track data is deleted.

This is needed because we can now delete information that is currently
cached in a trackModel.

kain88-de commented:
I changed the signals to be fired from trackdao to bstm directly.
